crypto/bio/bio_lib.c

```diff
@@ -796,31 +796,32 @@ static int bio_wait(BIO *bio, time_t max_time, unsigned int milliseconds)
         return 1;
```

Comment:
I wonder why the condition above this line `if (max_time == 0) `
makes sense, and why the return value is not 0 (timeout) in this case.

Reply:
As written in the documentation of `BIO_socket_wait()`, `BIO_wait()`, and `BIO_connect_retry()`, the meaning of `max_time == 0` is that no timeout is wanted (i.e., blocking behavior has been selected).
This, as well as the 'success' return in this case, is for convenient use for instance by `BIO_connect_retry()` and the HTTP client, where `max_time` has the same semantics.